syntax = "proto3";

package bot_builder_nlp.service;

service BotBuilderNlpService {
  rpc LoadBotStory(LoadBotStoryRequest) returns (LoadBotStoryResponse);
  rpc GetBotResponses(GetBotResponsesRequest) returns (GetBotResponsesResponse);
  rpc UpsertEmbedding(UpsertEmbeddingRequest) returns (UpsertEmbeddingResponse);
  rpc DeleteEmbedding(DeleteEmbeddingRequest) returns (DeleteEmbeddingResponse);
}

message LoadBotStoryRequest {
  string user_id = 1;
}

message LoadBotStoryResponse {
  repeated BotResponse responses = 1;
}

message GetBotResponsesRequest {
  string user_id = 1;
  string user_input = 2;
}

message GetBotResponsesResponse {
  repeated BotResponse responses = 1;
}

message BotResponseButton {
  string content = 1;
  string go_to = 2;
}

message BotResponseGalleryItem {
  string img_url = 1;
  string title = 2;
  string description = 3;
  repeated BotResponseButton buttons = 4;
}

message BotResponse {
  string type = 1;
  repeated string variants = 2;
  string img_url = 3;
  repeated BotResponseButton buttons = 4;
  repeated BotResponseGalleryItem gallery = 5;
}

message UserInput {
  string id = 1;
  string content = 2;
  string story_block_id = 3;
}

message UpsertEmbeddingRequest {
  string user_id = 1;
  repeated UserInput user_inputs = 2;
}

message UpsertEmbeddingResponse {
  bool success = 1;
}

message DeleteEmbeddingRequest {
  repeated string ids = 1;
}

message DeleteEmbeddingResponse {
  bool success = 1;
}